lang_interface/java/com/intel/daal/utils/LibUtils.java

```diff
@@ -48,24 +51,40 @@
      */
     public static void loadLibrary()
     {
-        try {
-            logger.log(logLevel, "Loading library " + DAALLIB + " as file.");
-            System.loadLibrary(DAALLIB);
-            logger.log(logLevel, "DONE: Loading library " + DAALLIB + " as file.");
-            return;
-        }
-        catch (UnsatisfiedLinkError e) {
-            logger.log(logLevel, "Can`t find library " + DAALLIB + " in java.library.path.");
-        }
+	if(loaded){
+		return;
+	}
+        synchronized(LibUtils.class){
+            if(loaded){
+                return;
+            }
+            String v = System.getProperty("tbb-threads", "1");
```

Comment:
you are going to change default threading behavior. now it use maximum num of threads, after your change - it will be one thread only. it will cause perf degradation for all current users

Reply:
good catch. I'll change the default value to number of CPU cores.